v3/src/components/graph/components/dotplotdots.tsx

```diff
@@ -211,41 +215,100 @@ export const DotPlotDots = observer(function DotPlotDots(props: PlotProps) {
           }
         }
 
-      const
-        // Note that we can get null for either or both of the next two functions. It just means that we have
-        // a circle for the case, but we're not plotting it.
-        getPrimaryScreenCoord = (anID: string) => {
-          const primaryCoord = primaryAxisScale(dataset?.getNumeric(anID, primaryAttrID) ?? -1) /
-              numExtraPrimaryBands,
-            extraPrimaryValue = dataset?.getStrValue(anID, extraPrimaryAttrID),
-            extraPrimaryCoord = extraPrimaryValue
-              ? extraPrimaryAxisScale(extraPrimaryValue ?? '__main__') ?? 0
-              : 0
-          return primaryCoord + extraPrimaryCoord
-        },
-        getSecondaryScreenCoord = (anID: string) => {
-          const secondaryCat = binMap[anID].category,
-            extraSecondaryCat = binMap[anID].extraCategory,
-            indexInBin = binMap[anID].indexInBin,
-            onePixelOffset = primaryIsBottom ? -1 : 1 // Separate circles from axis line by 1 pixel
-          return binMap[anID]
-            ? computeSecondaryCoord({secondaryCat, extraSecondaryCat, indexInBin}) + onePixelOffset
-            : null
-        },
-        getScreenX = primaryIsBottom ? getPrimaryScreenCoord : getSecondaryScreenCoord,
-        getScreenY = primaryIsBottom ? getSecondaryScreenCoord : getPrimaryScreenCoord,
-        getLegendColor = dataConfiguration?.attributeID('legend')
-          ? dataConfiguration?.getLegendColorForCase : undefined
+      const computePrimaryScreenCoord = (anID: string) => {
+        const primaryCoord = primaryAxisScale(dataset?.getNumeric(anID, primaryAttrID) ?? -1) / numExtraPrimaryBands
+        const extraPrimaryValue = dataset?.getStrValue(anID, extraPrimaryAttrID)
+        const extraPrimaryCoord = extraPrimaryValue ? extraPrimaryAxisScale(extraPrimaryValue ?? '__main__') ?? 0 : 0
+        return { primaryCoord, extraPrimaryCoord }
+      }
+
+      const subPlotKey = (anID: string) => {
+        const category = dataset?.getStrValue(anID, secondaryAttrID) ?? '__main__'
+        const extraCategory = dataset?.getStrValue(anID, extraSecondaryAttrID) ?? '__main__'
+        const extraPrimaryCategory = dataset?.getStrValue(anID, extraPrimaryAttrID) ?? '__main__'
+        const key: Record<string, string> = {}
+        secondaryAttrID && (key[secondaryAttrID] = category)
+        extraSecondaryAttrID && (key[extraSecondaryAttrID] = extraCategory)
+        extraPrimaryAttrID && (key[extraPrimaryAttrID] = extraPrimaryCategory)
+        return key
+      }
+
+      const computeBarDimension = (anID: string) => {
+        // The returned value is the width or height of the plot (depending the primary axis) divided by the
+        // number of cases in the same subplot as the case whose ID matches anID.
+        const subPlotCaseCount = dataConfig?.subPlotCases(subPlotKey(anID)).length ?? 0
```

Comment:
I'm not sure this use of `subPlotCases` is the best thing to do, but if so, maybe the related `subPlotKey` function above should be made a utility function or a view that could be used elsewhere? It seemed like the obvious thing to do to me coming from the adornment work. I think it'd be possible to get the same information, though, without using `subPlotCases` with something like the below which would be more in keeping with the existing code in this component.

```
const subPlotCaseCount = dataConfig?.caseDataArray.filter((c: CaseData) => {
           const cCat = dataset?.getStrValue(c.caseID, secondaryAttrID) ?? '__main__'
           const cExtraCat = dataset?.getStrValue(c.caseID, extraSecondaryAttrID) ?? '__main__'
           const cExtraPrimaryCat = dataset?.getStrValue(c.caseID, extraPrimaryAttrID) ?? '__main__'
           return cCat === category && cExtraCat === extraCategory && cExtraPrimaryCat === extraPrimaryCategory
         }).length ?? 0
```